### PR TITLE
style: Set root font-size right after cascading font-size

### PR DESCRIPTION
**Pull request by emilio**
This way rem in the root element works as expected.

This fixes layout/reftests/css-valuesandunits/unit-rem-root-width.html in Gecko.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16855)

### CODE REVIEW

File: components/style/properties/properties.mako.rs [hunk 1]
```diff
@@ -2665,6 +2663,14 @@ pub fn apply_declarations<'a, F, I>(device: &Device,
                                                  error_reporter);
             % endif
             }
+
+            let writing_mode = get_writing_mode(context.style.get_inheritedbox());
+            context.style.writing_mode = writing_mode;
```
Comment (upsuper) [inline components/style/properties/properties.mako.rs]: Why do you need to move writing mode down here? It doesn't seem to be related, and I suspect whether this is a right thing to do... IIRC font computation somehow depends on writing-mode, although I forget how :/

Comment (upsuper) [inline components/style/properties/properties.mako.rs]: Checking the spec, I believe `writing-mode` actually should be set before `font-size`, because font size may depend on that property when any font-relative length unit is used on it.

Comment (emilio) [inline components/style/properties/properties.mako.rs]: Yeah, but note that that writing mode is not the writing mode property itself. Anyway you're right, and I should probably leave it where it was..

Comment (emilio) [inline components/style/properties/properties.mako.rs]: Done
